python/cudf/cudf/core/series.py

```diff
@@ -181,22 +182,16 @@ class _SeriesIlocIndexer(_FrameIndexer):
     For integer-location based selection.
     """
 
+    _frame: cudf.Series
+
     @_cudf_nvtx_annotate
     def __getitem__(self, arg):
-        if isinstance(arg, tuple):
-            arg = list(arg)
-        data = self._frame._get_elements_from_column(arg)
-
-        if (
-            isinstance(data, (dict, list))
-            or _is_scalar_or_zero_d_array(data)
-            or _is_null_host_scalar(data)
-        ):
-            return data
-        return self._frame._from_data(
-            {self._frame.name: data},
-            index=cudf.Index(self._frame.index[arg]),
+        tag, key = indexing_utils.normalize_row_iloc_indexer(
```

Comment:
Would it be better to make a named tuple for `(tag, key)`?

Reply:
Probably, then I can change the `_get` signature to just take a single object.

So probably

```
class ParsedIndexer(NamedTuple):
    tag: IndexTag
    key: slice | NumericalColumn # Probably that will cause too many issues.
```

Reply:
Used dataclasses so I can add type annotations the members.